services/ImportManager.php

```diff
@@ -0,0 +1,491 @@
+        $dataStr = file_get_contents($apiUrl, false, $context);
+        return json_decode($dataStr, true);
+    }
+
+    /**
+     * Get the local path to files uploads (usually "files")
+     *
+     * @return string local path to files uploads
+     */
+    private function getLocalFileUploadPath()
+    {
+        if ($this->uploadPath !== null) {
+            return $this->uploadPath;
+        }
+
+        $attachConfig = $this->wiki->GetConfigValue("attach_config");
+
+        if (!is_array($attachConfig)) {
+            $attachConfig = array();
+        }
+
+        if (empty($attachConfig['upload_path'])) {
+            $this->uploadPath = 'files';
+        } else {
+            $this->uploadPath = $attachConfig['upload_path'];
+        }
+
+        return $this->uploadPath;
+    }
+
+    /**
+     * Download file url to local wiki using cURL
+     *
+     * @param string $from file url
+     * @param string $to local path
+     * @param boolean $overwrite overwrite existing file ? (default:false)
+     * @return void
+     */
+    private function cURLDownload($from, $to, $overwrite = false)
+    {
+        $output = '';
+        if (file_exists($to)) {
+            if ($overwrite) {
+                $output .= _t('LMS_FILE').' '.$to.' '._t('LMS_FILE_OVERWRITE').'.';
+            } else {
+                $output .= _t('LMS_FILE').' '.$to.' '._t('LMS_FILE_NO_OVERWRITE').'.';
+                return $output;
+            }
+        }
+
+        // Do cURL transfer
+        $fp = fopen($to, 'wb');
+        $ch = curl_init($from);
+        curl_setopt($ch, CURLOPT_FILE, $fp);
+        curl_setopt($ch, CURLOPT_HEADER, 0);
+        curl_setopt($ch, CURLOPT_FAILONERROR, true);
+        curl_exec($ch);
+        $err = curl_error($ch);
+        curl_close($ch);
+        fclose($fp);
+
+        if ($err) {
+            unlink($to);
+            throw new \Exception($output._t('LMS_ERROR_DOWNLOADING').' '.$filename.': '.$err."\n"._t('LMS_REMOVING_CORRUPTED_FILE').' '.$filename);
+        }
+        return $output;
+    }
+
+    /**
+     * Return fields that may contain attachments to import (body for wikipage, or textelong fields for bazar entries)
+     *
+     * @param array $wikiPage page or entry content as an array
+     * @return array keys of $wikiPage that may contain attachments to import
+     */
+    public function getTextFieldsFromWikiPage($wikiPage)
+    {
+        $fields= [];
+        if (!empty($wikiPage['tag'])) { // classic wiki page
+            $fields[] ='body';
+        } elseif (!empty($wikiPage['id_fiche'])) { // bazar entry
+            $formManager = $this->wiki->services->get(FormManager::class);
+            $form = $formManager->getOne($wikiPage['id_typeannonce']);
+            // find fields that are textareas
+            foreach ($form['prepared'] as $field) {
+                if (get_class($field) === 'YesWiki\Bazar\Field\TextareaField') {
```

Comment:
```suggestion
                if ($field instanceof TextareaField) {
```

me semble plus pratique pour pouvoir aussi importer les fields custom qui pourraient hériter de `TextareaField`.
Sauf si vous voulez absolument restreindre aux TextareaField

Reply:
oui j'avais tenté cela, mais il me sortais `false`..  
Ca ne marche que si je mets aussi `use YesWiki\Bazar\Field\TextareaField;` ?

Reply:
Effectivement, si tu mets `($field instanceof TextareaField)` ça rendra faux car l'interpréteur php ne trouve pas la définition de la classe `TextareaField` dans le dossier courant (ou dans tous les dossiers de `path`) .
Pour que ça fonctionne, soit tu mets `($field instanceof YesWiki\Bazar\Field\TextareaField)` soit tu définis l'alias
`TextareaField = YesWiki\Bazar\Field\TextareaField` grâce à la ligne `use YesWiki\Bazar\Field\TextareaField;`
et alors tu peux écrire `($field instanceof TextareaField)`. L'interpréteur comprend que `TextareaField` est la la classe `TextareaField` du namespace `YesWiki\Bazar\Field\`